message/injector.cpp

```diff
@@ -1581,6 +1581,13 @@ void Injector::insertMessages()
 
         // Now we insert the headers and bodies of every MIME bodypart.
 
+        Bodypart *bp;
+        if ( m->hasPGPsignedPart() ) {
+            EString pnr( "raw-pgp-signed" );
```

Comment:
The parent may not be the one with c-t multipart/signed.
Any subtree of the MIME tree may be signed.